### ISSUE TITLE
Update echoserv_ssl.py to use react

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7083]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#7083|
|Type|enhancement|
|Created|2014-04-06 12:32:36Z|
|Branch|https://github.com/twisted/twisted/tree/echoserv-ssl-react-7083|

"Single shot" clients like this one - which perform some well-defined operation with a clear completion condition - should use `twisted.internet.task.react`.

Attachments:

* [echoserv-react.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/4de/4dea86950d0908f10bb5d6b3481f77c06b1be79f/772d10ab2769ea7de62bc540e826ee412abd5590.patch) (4227 bytes) - added by mythmon on 2014-04-14 20:07:35Z - 
* [echoserv-react.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/4de/4dea86950d0908f10bb5d6b3481f77c06b1be79f/889b3959381688d29c411e952121604afc64bb0c.patch) (3795 bytes) - added by mythmon on 2014-04-16 15:02:19Z - Version 2

<details><summary>Searchable metadata</summary>

```
trac-id__7083 7083
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_echoserv_ssl_react_7083 branches/echoserv-ssl-react-7083
branch_author__hynek hynek
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1396787556000000 1396787556000000
changetime__1413209064014489 1413209064014489
version__None None
owner__hawkowl hawkowl
cc__mythmon
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="mythmon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|mythmon set owner to mythmon<br>mythmon set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="mythmon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|mythmon commented|
|-|-|

Replying to [exarkun](#7083):
> "Single shot" clients like this one - which perform some well-defined operation with a clear completion condition - should use `twisted.internet.task.react`.

In the subject you said "Update echoserv_ssl.py", but in the description you said "clients like this". Are you talking about "echoclient_ssl.py" instead?

**Comment by twisted-trac** [comment]
|<img alt="mythmon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|mythmon removed owner<br>mythmon set status to `new`|
|-|-|

Assuming that the point of this was to fix the clients, I did that.

**Comment by twisted-trac** [comment]
|[<img alt="habnabit's avatar" src="https://avatars.githubusercontent.com/u/21168?s=50" width="50" height="50">](https://github.com/habnabit)<a name="note_4"></a>|@habnabit set owner to mythmon|
|-|-|

Thanks for the contribution! There's a few minor issues with it.

 1. You appear to have an extra bit in your diff that's not related to the fix, patching _sslverify.
 2. [All changes need a news file.](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles)
 3. You call `self.done.errback(None)`, which will fail, because there is no current exception when `clientConnectionFailed` is called. You might mean instead `self.done.callback(None)` or `self.done.errback(reason)`

**Comment by twisted-trac** [comment]
|<img alt="mythmon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|mythmon removed owner|
|-|-|

Fixed issues pointed out in comment 4.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

Latest patch looks good to merge. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_7"></a>|@hynek commented|
|-|-|

(In [43346]) Branching to echoserv-ssl-react-7083.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_8"></a>|@hynek commented|
|-|-|

(In [43350]) Apply mythmon's patch

At least the parts that still apply, echoclient_ssl.py is using react already.

refs [#7083](https://github.com/twisted/twisted/issues/7083)

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_9"></a>|@hynek set owner to @hawkowl|
|-|-|

Turns out™ echoclient_ssl.py has already been ported over, but echoclient.py itself not.

So I applied what worked and added some prettifications to it to adhere to our standards.  Keeping it in review, so someone (an owl volunteered) can double-check my changes.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl commented|
|-|-|

LGTM, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_11"></a>|@hynek set status to `closed`|
|-|-|

(In [43355]) Merge echoserv-ssl-react-7083

Author: mythmon
Reviewers: habnabit, hawkowl
Fixes: [#7083](https://github.com/twisted/twisted/issues/7083)

The echoclient example now uses twisted.internet.task.react.
